### PR TITLE
Create Unit Tests for Header Extraction  to IteratorType and SeqType

### LINKED ISSUE #30: Create Unit Tests for Header Extraction for each type

### PR DESCRIPTION
**Pull request by ricardochaves9**
Create Unit Tests for Header Extraction #30 

right now added:

**IteratorType**  CSV - completed, PSV - completed ; TSV - completed

**SeqType** CSV completed, PSV - completed ; TSV - completed

### CODE REVIEW

File: src/test/scala/Tests.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers._
+
+
+class Tests extends AnyFlatSpec {
+
+
+
+  "iterator" should "return header and line" in{
```
Comment (PedroCorreiaLuis) [inline src/test/scala/Tests.scala]: You are actually testing the Scala Scala code and not our code, you need to test our code

File: src/test/scala/parsing/extractHeadersTests.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    val source: Iterator[String] = Iterator("Ballon dor","Karim Benzema")
+
+    val parseInputSource: ParserInput = ParserInput(
+      in = IteratorType(source),
+      csvDefinition = formats.CSV,
+      headers = Nil
+    )(Parser.encode)
+
+    val getHeader: ParserInput = extractHeaders(parseInputSource)
+    getHeader.headers shouldBe source.toList.take(1)
```
Comment (joao-penim) [inline src/test/scala/parsing/extractHeadersTests.scala]: here I recommend not using `take(1)` but actually testing directly against what you are expecting, now you're just testing `take(1)` against a `take(1)`

```suggestion
    getHeader.headers shouldBe Iterator("Ballon dor")
```

This applies to the other tests as well

**Review by joao-penim — DISMISSED** [review]

File: src/test/scala/parsing/extractHeadersTests.scala (hunk 2 above)
Comment (ricardochaves9) [reply src/test/scala/parsing/extractHeadersTests.scala]: thanks for your recommendation, already done

File: src/test/scala/parsing/extractHeadersTests.scala [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+    val source1: Seq[String] = Seq("Ballon dor\n\n\n\n\n,,,\\\\\\\\,,Ballon dor feminin,Kopa Trophy,\r\n,,\r\n,Yashing Trophy,,,,\n\n\n\n\r\n\r\n","Karim Benzema, Alexia Putellas, Gavi, Courtois")
+    val parseInputSource1: ParserInput = ParserInput(
+      in = SeqType(source1),
+      csvDefinition = formats.CSV,
+      headers = Nil
+    )(Parser.encode)
+
+    val getHeader: ParserInput = extractHeaders(parseInputSource)
+    getHeader.headers shouldBe Seq("Ballon dor\n\n\n\n\n,,,\\\\\\\\,,Ballon dor feminin,Kopa Trophy,\r\n,,\r\n,Yashing Trophy,").toList
```
Comment (joao-penim) [inline src/test/scala/parsing/extractHeadersTests.scala]: Just a small thing: You can create a `List(...)` directly instead of `Seq(...).toList`

Comment (ricardochaves9) [reply src/test/scala/parsing/extractHeadersTests.scala]: thank you, done

File: src/test/scala/parsing/extractHeadersTests.scala [hunk 4]
```diff
@@ -0,0 +1,252 @@
+package parsing
+
+
+import models.{IteratorType, ParserInput, SeqType}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers._
+import parsing.Parser.extractHeaders
+
+class extractHeadersTests extends AnyFlatSpec {
```
Comment (PedroCorreiaLuis) [inline src/test/scala/parsing/extractHeadersTests.scala]:
```suggestion
class ExtractHeadersTests extends AnyFlatSpec {
```

File: src/test/scala/parsing/extractHeadersTests.scala [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+    )(Parser.encode)
+
+    val source1: Iterator[String] = Iterator("Ballon dor\n\n\n\n\n,,,\\\\\\\\,,Ballon dor feminin,Kopa Trophy,\r\n,,\r\n,Yashing Trophy,,,,\n\n\n\n\r\n\r\n","Karim Benzema, Alexia Putellas, Gavi, Courtois")
+    val parseInputSource1: ParserInput = ParserInput(
+      in = IteratorType(source1),
+      csvDefinition = formats.CSV,
+      headers = Nil
+    )(Parser.encode)
+
+    val source2  =Iterator("\"Ballon Dor List\",\"club\"\n\"Karim Benzema\", \"Real Madrid\"\n\"Sadio Mane\", \"Liverpool\"\n\"Kevin De Bruyne\", \"Manchester  City\"\n\"Robert Lewandowski\", \"Barcelona\"\n\"Mohamed Salah\", \"Liverpool\"\n\"Ricardo Esgaio\", \"Sporting\"\n\"Mbappe\", \"PSG\"")
```
Comment (PedroCorreiaLuis) [inline src/test/scala/parsing/extractHeadersTests.scala]: missing type and scalafmt here

File: src/test/scala/parsing/extractHeadersTests.scala [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+    )(Parser.encode)
+
+    val source1: Iterator[String] = Iterator("Ballon Dor List", "| club\nKarim Benzema"," | Real Madrid\nSadio Mane | Liverpool\nKevin De Bruyne | Manchester  City\nRobert Lewandowski | Barcelona\nMohamed Salah | Liverpool\nRicardo Esgaio | Sporting\nMbappe | PSG")
+    val parseInputSource1: ParserInput = ParserInput(
+      in = IteratorType(source1),
+      csvDefinition = formats.PSV,
+      headers = Nil
+    )(Parser.encode)
+
+    val source2  =Iterator("\"Ballon Dor List\"|\"club\"\n\"Karim Benzema\"|\"Real Madrid\"\n\"Sadio Mane\" | \"Liverpool\"\n\"Kevin De Bruyne\" | \"Manchester  City\"\n\"Robert Lewandowski\" | \"Barcelona\"\n\"Mohamed Salah\" | \"Liverpool\"\n\"Ricardo Esgaio\" | \"Sporting\"\n\"Mbappe\" | \"PSG\"")
```
Comment (PedroCorreiaLuis) [inline src/test/scala/parsing/extractHeadersTests.scala]: same here

File: src/test/scala/parsing/extractHeadersTests.scala [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+    )(Parser.encode)
+
+    val source1: Iterator[String] = Iterator("Ballon Dor List\t club","Karim Benzema\tReal Madrid\nSadio Mane\tLiverpool\nKevin De Bruyne\tManchester  City\nRobert Lewandowski\t Barcelona\nMohamed Salah\t Liverpool\nRicardo Esgaio\t Sporting\nMbappe\tPSG")
+    val parseInputSource1: ParserInput = ParserInput(
+      in = IteratorType(source1),
+      csvDefinition = formats.TSV,
+      headers = Nil
+    )(Parser.encode)
+
+    val source2  =Iterator("Ballon Dor List\t\t\t\tclub\r\n","Karim Benzema\t\t\t\tReal Madrid\nSadio Mane\tLiverpool\nKevin De Bruyne\tManchester  City\\\nRobert Lewandowski\t Barcelona\nMohamed Salah\t Liverpool\nRicardo Esgaio\t Sporting\nMbappe\tPSG")
```
Comment (PedroCorreiaLuis) [inline src/test/scala/parsing/extractHeadersTests.scala]: and here

File: src/test/scala/parsing/extractHeadersTests.scala [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+    )(Parser.encode)
+
+    val source1: Seq[String] = Seq("Ballon Dor List", "| club\nKarim Benzema"," | Real Madrid\nSadio Mane | Liverpool\nKevin De Bruyne | Manchester  City\nRobert Lewandowski | Barcelona\nMohamed Salah | Liverpool\nRicardo Esgaio | Sporting\nMbappe | PSG")
+    val parseInputSource1: ParserInput = ParserInput(
+      in = SeqType(source1),
+      csvDefinition = formats.PSV,
+      headers = Nil
+    )(Parser.encode)
+
+    val source2  =Seq("\"Ballon Dor List\"|\"club\"\n\"Karim Benzema\"|\"Real Madrid\"\n\"Sadio Mane\" | \"Liverpool\"\n\"Kevin De Bruyne\" | \"Manchester  City\"\n\"Robert Lewandowski\" | \"Barcelona\"\n\"Mohamed Salah\" | \"Liverpool\"\n\"Ricardo Esgaio\" | \"Sporting\"\n\"Mbappe\" | \"PSG\"")
```
Comment (PedroCorreiaLuis) [inline src/test/scala/parsing/extractHeadersTests.scala]: and here

File: src/test/scala/parsing/extractHeadersTests.scala [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+    )(Parser.encode)
+
+    val source1: Seq[String] = Seq("Ballon Dor List\t club","Karim Benzema\tReal Madrid\nSadio Mane\tLiverpool\nKevin De Bruyne\tManchester  City\nRobert Lewandowski\t Barcelona\nMohamed Salah\t Liverpool\nRicardo Esgaio\t Sporting\nMbappe\tPSG")
+    val parseInputSource1: ParserInput = ParserInput(
+      in = SeqType(source1),
+      csvDefinition = formats.TSV,
+      headers = Nil
+    )(Parser.encode)
+
+    val source2  =Seq("Ballon Dor List\t\t\t\tclub\r\n","Karim Benzema\t\t\t\tReal Madrid\nSadio Mane\tLiverpool\nKevin De Bruyne\tManchester  City\\\nRobert Lewandowski\t Barcelona\nMohamed Salah\t Liverpool\nRicardo Esgaio\t Sporting\nMbappe\tPSG")
```
Comment (PedroCorreiaLuis) [inline src/test/scala/parsing/extractHeadersTests.scala]: .

**Review by PedroCorreiaLuis — COMMENTED** [review]
Left some suggestions

**Review by PedroCorreiaLuis — APPROVED** [review]

**Pull request merged**